fix: wrong trigger of rollback after commit logic

danielskinstad opened the pull request:
Fixed an issue where rollback after commit was triggered during regular rollbacks because it only checked if a new composition was absent. Added a check for a previous composition, since that is cleaned up during commit.

Ticket: MEN-8976

vpodzime reviewed (commented):
Looks good to me. Can you adjust the existing tests to catch this issue if we reintroduce it?

danielskinstad committed:
fix: wrong trigger of rollback after commit logic

Fixed an issue where rollback after commit was triggered during regular
rollbacks because it only checked if a new composition was absent. Added
a check for a previous composition, since that is cleaned up during commit.

Ticket: MEN-8976

Signed-off-by: Daniel Skinstad Drabitzius <daniel.drabitzius@northern.tech>

danielskinstad committed:
test: test rollback with no 'new' directory

Test that we successfully rollback if `ArtifactInstall` failed to create
`new`. This means that there's only `previous` and it should
successfully rollback and restore it as `current`.

Ticket: MEN-8976

Signed-off-by: Daniel Skinstad Drabitzius <daniel.drabitzius@northern.tech>

vpodzime reviewed (approved):
I can't say I can fully comprehend the logic of all the possible cases here, but this looks legit to me.
